x/distribution/genesis.go

```diff
@@ -34,6 +38,25 @@ func InitGenesis(ctx sdk.Context, keeper Keeper, data types.GenesisState) {
 	for _, evt := range data.ValidatorSlashEvents {
 		keeper.SetValidatorSlashEvent(ctx, evt.ValidatorAddress, evt.Height, evt.Event)
 	}
+
+	moduleHoldings = moduleHoldings.Add(data.FeePool.CommunityPool)
+	moduleHoldingsInt, _ := moduleHoldings.TruncateDecimal()
+
+	// check if the module account exists and create it if not
+	moduleAcc := keeper.GetDistributionAccount(ctx)
+	if moduleAcc == nil {
+		moduleAcc = supply.NewModuleHolderAccount(types.ModuleName)
+		if err := moduleAcc.SetAccountNumber(ak.GetNextAccountNumber(ctx)); err != nil {
+			panic(err)
+		}
+	}
+
+	if moduleAcc.GetCoins().IsZero() {
+		if err := moduleAcc.SetCoins(moduleHoldingsInt); err != nil {
```

Comment:
what if someone doesn't import some of those modules? there'd be empty accounts that are not being used. @alexanderbez very much agreed with setting them on the app initialization instead of the supply `InitGenesis`.